### DISCUSSION TITLE
The files are not actually saved on NFS share

### DISCUSSION THREAD

**Discussion by lasselehtinen** [spatie/laravel-medialibrary#2012] [Q&A]
Hi,

I swapped an existing working configuration to use a shared NFS directory from storage_path('media'). This because I have multiple Horizon workers that need access to the files Everything seemed fine first, the media is indeed saved to the collection and it appears in the database. However when trying to access the file, it is not there. For some reason the file disappears in thin air and I cannot find the file anywhere in the file system. Any ideas how to debug this? I assume there are checks in place if saving the file fails for some reason. Any ideas what might be causing this? 

The media is added:

`$product->addMediaFromUrl($url)->withCustomProperties(['md5' => md5($contents)])->toMediaCollection('cover');`

```
>>> config('media-library.disk_name');
=> "mediafiles"
>>> config('filesystems.disks.mediafiles');
=> [
     "driver" => "local",
     "root" => "/mnt/medialibrary",
   ]
>>> $product = Product::where('gtin', 9789520416478)->first();
[!] Aliasing 'Product' to 'App\Product' for this Tinker session.
=> App\Product {#4523
     id: 5090,
     created_at: "2019-10-07 21:46:51",
     updated_at: "2020-08-22 20:08:53",
     publisher_id: 5,
     gtin: 9789520416478,
   }
>>> $product->getFirstMediaPath('cover');
=> "/mnt/medialibrary/18375/9789520416478_frontcover_final.jpg"
>>> file_exists($product->getFirstMediaPath('cover'));
=> false
>>>
```

I have tried:

- This is should be a permission issue, the mount is rw and I can write to the mount from all machines
- There are no SELinux warnings and disabling it completely does not make a difference
- Reverting back to storage_path('media') always fixes the problem

**Comment by francoism90** [comment]
Did you manage to solve it? If not, are you sure the user http (or any other process owner) is allowed to write to the NFS share? Do you see the NFS mount in `mount -a` or `df -h`?

I don't write to a NFS share, but I'm writing to another disk which seems to go fine.

**Reply by lasselehtinen** [reply]
The mount is visible with mount and I can write from artisan tinker. It should not be an SELinux issue either, I cannot see any warnings. Next I will try if the nginx/php-fpm processes have write access to the mount. I have a workaround by using the sftp filesystem for now. I will debug this more when I have more time.
